rasa/core/actions/forms.py

```diff
@@ -258,22 +258,15 @@ async def validate_slots(
         validate_name = f"validate_{self.name()}"
 
         if validate_name not in domain.action_names_or_texts:
-            return events
+            return []
```

Comment:
would also be good to add some docstr to the temporary tracker (or maybe there's also a better name? 🤔 ) -- wasn't  really aware that the validation action would not receive the original tracker  but just the slot set events plus some very selectively chosen events :D -- Think we even might need to add that to some docs. Wdyt?

Reply:
Yup, sure, will see what I can do about the docstring and if anything in Forms section of the docs should be updated too. I guess the idea with the temporary tracker is that we only want to look at SlotSet events from last user turn.

Reply:
Think that's a good guess and if that is the only reason, then I'd vote for removing that implicit restriction in the long run. Short term we should definitely explicitly state that in the docs because it could be very confusing  for people who write custom actions and expect the tracker to contain the same things they see when they implement a custom rule policy :/.

Reply:
I've extended a sentence in the `Validating Form Input` docs section, that seemed the only appropriate place to clarify that only recent `SlotSet` events are validated. The style of the docs is not to go too heavy on internal implementation details, so I hope this is good enough, a paragraph on some temporary tracker used when running custom action could be a bit confusing for new users.